File: backend/models/order.py
# BoxDesign AI — Order Model
from sqlalchemy import Column, String, DateTime, Integer, ForeignKey, Enum
from sqlalchemy.orm import relationship
from models.user import Base
from datetime import datetime
import uuid
import enum

class OrderStatus(str, enum.Enum):
    DRAFT = "draft"
    APPROVED = "approved"
    PAID = "paid"
    GENERATING = "generating"
    DELIVERED = "delivered"
    FAILED = "failed"

class PricingTier(str, enum.Enum):
    BASIC = "basic"
    STANDARD = "standard"
    PREMIUM = "premium"

class Order(Base):
    """
    Tracks customer orders, payments, and the delivery status of final production files.
    """
    __tablename__ = "orders"

    id = Column(String, primary_key=True, default=lambda: str(uuid.uuid4()))
    user_id = Column(String, ForeignKey("users.id"), nullable=False, index=True)
    design_request_id = Column(String, nullable=True)
    selected_design_id = Column(String, nullable=True)
    
    # Tier and Status
    pricing_tier = Column(Enum(PricingTier), default=PricingTier.STANDARD)
    status = Column(Enum(OrderStatus), default=OrderStatus.DRAFT)
    
    # Pricing Details (in Paise or Rupees, usually Integer for precision)
    base_amount_inr = Column(Integer, nullable=False)
    gst_amount_inr = Column(Integer, nullable=False)
    total_amount_inr = Column(Integer, nullable=False)
    promo_code = Column(String(50), nullable=True)
    discount_inr = Column(Integer, default=0)
    
    # Payment Tracking
    razorpay_order_id = Column(String(255), nullable=True)
    razorpay_payment_id = Column(String(255), nullable=True)
    
    # Approvals
    approved_by_name = Column(String(100), nullable=True)
    approved_at = Column(DateTime, nullable=True)
    
    # Final Asset URLs
    pdf_url = Column(String(500), nullable=True)
    png_url = Column(String(500), nullable=True)
    cdr_url = Column(String(500), nullable=True) # Source CDR or high-res SVG
    invoice_url = Column(String(500), nullable=True)
    files_expire_at = Column(DateTime, nullable=True)
    
    # Timestamps
    created_at = Column(DateTime, default=datetime.utcnow)
    updated_at = Column(DateTime, default=datetime.utcnow, onupdate=datetime.utcnow)

    # Relationships
    user = relationship("User", backref="orders")

File: src/types/order.ts
// ─── BoxDesign AI — Order & Payment Types ────────────────────────────────────

export type OrderStatus  = 'draft' | 'approved' | 'paid' | 'generating' | 'delivered' | 'failed';
export type PricingTier  = 'basic' | 'standard' | 'premium';
export type FileType     = 'pdf' | 'png' | 'cdr';

export interface PricingPlan {
  tier:        PricingTier;
  label:       string;
  priceInr:    number;
  files:       FileType[];
  revisions:   number;
  description: string;
}

export const PRICING_PLANS: PricingPlan[] = [
  {
    tier: 'basic',
    label: 'Basic',
    priceInr: 299,
    files: ['pdf'],
    revisions: 0,
    description: 'PDF flat design only',
  },
  {
    tier: 'standard',
    label: 'Standard',
    priceInr: 799,
    files: ['pdf', 'png', 'cdr'],
    revisions: 1,
    description: 'PDF + PNG + CDR, 1 revision',
  },
  {
    tier: 'premium',
    label: 'Premium',
    priceInr: 1499,
    files: ['pdf', 'png', 'cdr'],
    revisions: 3,
    description: 'All files + source layers + 3 revisions + priority support',
  },
];

export interface Order {
  id: string;
  userId: string;
  designRequestId: string;
  selectedDesignId: string;
  pricingTier: PricingTier;
  status: OrderStatus;
  totalAmountInr: number;
  gstAmountInr: number;
  promoCode?: string;
  discountInr?: number;
  paymentId?: string;
  approvedByName?: string;
  approvedAt?: string;
  files?: DeliveredFile[];
  createdAt: string;
  updatedAt: string;
}

export interface DeliveredFile {
  type: FileType;
  url: string;         // Signed URL (expires in 90 days)
  sizeBytes: number;
  expiresAt: string;
}

export interface PaymentIntent {
  orderId: string;
  razorpayOrderId: string;
  amount: number;      // in paise
  currency: 'INR';
  keyId: string;
}
